prowler/providers/aws/aws_provider.py

```diff
@@ -134,6 +153,27 @@ def __init__(
             - ArgumentTypeError: If the input external ID is invalid.
             - ArgumentTypeError: If the input role session name is invalid.
 
+        Usage:
+            - Using static credentials:
+                provider = AwsProvider(
+                    aws_access_key_id="AWS_ACCESS_KEY_ID",
+                    aws_secret_access_key="AWS_SECRET_ACCESS_KEY",
+                    aws_session_token="AWS_SESSION_TOKEN",
+                )
+            - Using a profile:
+                provider = AwsProvider(profile="PROFILE_NAME")
+            - Using an IAM role:
+                provider = AwsProvider(role_arn="ROLE_ARN")
+            - Using an IAM role and setting session duration with external ID:
+                provider = AwsProvider(
+                    role_arn="ROLE_ARN",
+                    session_duration=3600,
+                    external_id="EXTERNAL_ID",
+                )
+            - Using an IAM role and MFA:
+                provider = AwsProvider(role_arn="ROLE_ARN", mfa=True)
+            - Using role session name:
+                provider = AwsProvider(role_arn="ROLE_ARN", role_session_name="ROLE_SESSION_NAME")
```

Comment:
In this block we need to also add the following:
- We use `boto3` so we follow their credential setup process.
- We need to split between authentication, getting credentials with permissions in AWS, and assuming a role.
- `static credentials` and `session credentials` because they are different if static from an IAM user or dynamic from the SSO.
- More details about the `mfa` since it is interactive at this time.
- Role session name format too.
- You can choose between documenting the SDK auth options or link to boto3 since the profile/config can also have defaults.

Please cover all the possibilities here.

The examples section is great but we need to add more detailed documentation to all the authentication methods, and the assume role provider.